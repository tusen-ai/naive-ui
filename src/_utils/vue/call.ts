type AnyFunction = (...args: any[]) => any

function call (funcs: MaybeArray<() => void>): void
function call<A1> (funcs: MaybeArray<(a1: A1) => void>, a1: A1): void
function call<A1, A2> (
  funcs: MaybeArray<(a1: A1, a2: A2) => void>,
  a1: A1,
  a2: A2
): void
function call<A1, A2, A3> (
  funcs: MaybeArray<(a1: A1, a2: A2, a3: A3) => void>,
  a1: A1,
  a2: A2,
  a3: A3
): void
function call<A1, A2, A3, A4> (
  funcs: MaybeArray<(a1: A1, a2: A2, a3: A3, a4: A4) => void>,
  a1: A1,
  a2: A2,
  a3: A3,
  a4: A4
): void
function call<A extends any[]> (
  funcs: AnyFunction[] | AnyFunction,
  ...args: A
): void {
  if (Array.isArray(funcs)) {
    funcs.forEach((func) => (call as any)(func, ...args))
  } else return funcs(...args)
}

export { call }

export type MaybeArray<T> = T | T[]
