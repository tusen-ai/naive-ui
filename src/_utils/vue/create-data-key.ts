export function createDataKey(key: string | number): string {
  return typeof key === 'string' ? `s-${key}` : `n-${key}`
}
