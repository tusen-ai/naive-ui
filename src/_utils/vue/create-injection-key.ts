import type { InjectionKey } from 'vue'

export function createInjectionKey<T>(key: string): InjectionKey<T> {
  return key as any
}
