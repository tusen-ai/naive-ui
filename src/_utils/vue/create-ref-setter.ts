import type { Ref } from 'vue'

export function createRefSetter (ref: Ref<HTMLElement | null>): any {
  return (inst: { $el: HTMLElement | null } | null) => {
    if (inst) {
      ref.value = inst.$el
    } else {
      ref.value = null
    }
  }
}
