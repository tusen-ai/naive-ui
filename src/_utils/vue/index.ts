export { getSlot } from './get-slot'
export { getVNodeChildren } from './get-v-node-children'
export { keep } from './keep'
export { omit } from './omit'
export { flatten } from './flatten'
export { call } from './call'
export { keysOf } from './keysOf'
export { render } from './render'
export { getFirstSlotVNode } from './get-first-slot-vnode'
export { createDataKey } from './create-data-key'
export { createRefSetter } from './create-ref-setter'
export { createInjectionKey } from './create-injection-key'
export {
  resolveSlot,
  resolveWrappedSlot,
  resolveSlotWithProps,
  resolveWrappedSlotWithProps,
  isSlotEmpty
} from './resolve-slot'
export type { MaybeArray } from './call'
export { mergeEventHandlers } from './merge-handlers'
export { isNodeVShowFalse } from './is-node-v-show-false'
export { Wrapper } from './wrapper'
