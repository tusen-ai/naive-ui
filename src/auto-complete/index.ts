export { autoCompleteProps, default as NAutoComplete } from './src/AutoComplete'
export type { AutoCompleteProps, AutoCompleteSlots } from './src/AutoComplete'
export type {
  AutoCompleteDefaultSlotProps,
  AutoCompleteGroupOption,
  AutoCompleteInst,
  AutoCompleteOption
} from './src/interface'
