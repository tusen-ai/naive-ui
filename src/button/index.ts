export {
  buttonProps,
  default as NButton,
  XButton as NxButton
} from './src/Button'
export type { ButtonProps } from './src/Button'
