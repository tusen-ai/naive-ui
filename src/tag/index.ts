export { default as NTag, tagProps } from './src/Tag'
export type { TagProps, TagSlots } from './src/Tag'
