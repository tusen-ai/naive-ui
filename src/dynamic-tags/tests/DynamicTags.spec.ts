import { mount } from '@vue/test-utils'
import { h } from 'vue'
import { NDynamicTags } from '../index'
import { NAutoComplete } from '../../auto-complete/index'
import { NButton } from '../../button/index'
import { NTag } from '../../tag'

describe('n-dynamic-tags', () => {
  it('should work with import on demand', () => {
    mount(NDynamicTags)
  })

  it('should work with `closable` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        defaultValue: ['教师', '程序员']
      }
    })
    expect(wrapper.find('.n-tag__close').exists()).toBe(true)
    await wrapper.setProps({ closable: false })
    expect(wrapper.find('.n-tag__close').exists()).toBe(false)
    wrapper.unmount()
  })

  it('should work with `color` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        color: {
          color: '#ccc',
          textColor: '#555',
          borderColor: 'rgb(85, 85, 85)'
        },
        defaultValue: ['教师', '程序员']
      }
    })
    expect(wrapper.find('.n-tag').attributes('style')).toContain(
      '--n-color: #ccc;'
    )
    expect(wrapper.find('.n-tag').attributes('style')).toContain(
      '--n-text-color: #555;'
    )
    expect(wrapper.find('.n-tag__border').attributes('style')).toContain(
      'border-color: rgb(85, 85, 85);'
    )
    wrapper.unmount()
  })

  it('should work with `disabled` prop', async () => {
    const onClose = vi.fn()
    const wrapper = mount(NDynamicTags, {
      props: {
        disabled: true,
        closable: true,
        defaultValue: ['教师', '程序员'],
        onClose
      }
    })

    expect(wrapper.find('.n-tag').classes()).toContain('n-tag--disabled')
    wrapper.find('.n-tag__close').trigger('click')
    expect(onClose).not.toBeCalled()
    expect(wrapper.find('.n-button').classes()).toContain('n-button--disabled')
    wrapper.unmount()
  })

  it('should work with `max` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        max: 2,
        value: ['教师']
      }
    })

    expect(wrapper.find('.n-button').classes()).not.toContain(
      'n-button--disabled'
    )
    await wrapper.setProps({ value: ['教师', '程序员'] })
    expect(wrapper.find('.n-button').classes()).toContain('n-button--disabled')
  })

  it('should work with `round` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        round: true,
        defaultValue: ['教师']
      }
    })
    expect(wrapper.find('.n-tag').classes()).toContain('n-tag--round')
    await wrapper.setProps({ round: false })
    expect(wrapper.find('.n-tag').classes()).not.toContain('n-tag--round')
    wrapper.unmount()
  })

  it('should work with `size` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        size: 'small',
        defaultValue: ['教师']
      }
    })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()
    await wrapper.setProps({ size: 'medium' })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()
    await wrapper.setProps({ size: 'large' })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with `type` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        size: 'small',
        defaultValue: ['教师']
      }
    })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ type: 'info' })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ type: 'success' })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ type: 'warning' })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ type: 'error' })
    expect(wrapper.find('.n-tag').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })

  it('should work with `tag-style` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        tagStyle: {
          color: 'rgb(79, 178, 51)'
        },
        defaultValue: ['教师']
      }
    })
    expect(wrapper.find('.n-tag').attributes('style')).toContain(
      'color: rgb(79, 178, 51);'
    )
    await wrapper.setProps({ tagStyle: { width: '100px' } })
    expect(wrapper.find('.n-tag').attributes('style')).toContain(
      'width: 100px;'
    )
    wrapper.unmount()
  })

  it('should work with `on-update:value` prop', () => {
    const onUpdateValue = vi.fn()
    const wrapper = mount(NDynamicTags, {
      props: {
        value: ['教师', '程序员'],
        onUpdateValue
      }
    })
    wrapper.find('.n-tag__close').trigger('click')
    expect(onUpdateValue).toBeCalled()
    wrapper.unmount()
  })

  it('should work with `input-props` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        defaultValue: ['教师']
      }
    })
    await wrapper.find('.n-button').trigger('click')
    expect(wrapper.find('.n-input').classes()).not.toContain(
      'n-input--disabled'
    )

    await wrapper.setProps({ inputProps: { disabled: true } })
    expect(wrapper.find('.n-input').classes()).toContain('n-input--disabled')
    wrapper.unmount()
  })

  it('should work with `input-style` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        defaultValue: ['教师']
      }
    })
    await wrapper.find('.n-button').trigger('click')
    expect(wrapper.find('.n-input').attributes('style')).not.toContain(
      'color: red'
    )

    await wrapper.setProps({ inputStyle: { color: 'red' } })
    expect(wrapper.find('.n-input').attributes('style')).toContain('color: red')
    wrapper.unmount()
  })

  it('should work with `render-tag` prop', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        value: ['教师'],
        renderTag: (tag: string) =>
          h(NTag, null, {
            default: () => `test-${tag}`
          })
      }
    })

    expect(wrapper.find('.n-tag__content').text()).toContain('test-教师')
    wrapper.unmount()
  })

  it('should work with `input` slot', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        value: ['教师', '程序员']
      },
      slots: {
        input: () => h(NAutoComplete)
      }
    })
    await wrapper.find('.n-button').trigger('click')
    expect(wrapper.find('.n-auto-complete').exists()).toBe(true)
    wrapper.unmount()
  })

  it('should work with `trigger` slot', async () => {
    const wrapper = mount(NDynamicTags, {
      props: {
        value: ['教师', '程序员']
      },
      slots: {
        trigger: () =>
          h(NButton, null, {
            default: () => '添加'
          })
      }
    })
    expect(wrapper.find('.n-button__content').text()).toEqual('添加')
    wrapper.unmount()
  })
})
