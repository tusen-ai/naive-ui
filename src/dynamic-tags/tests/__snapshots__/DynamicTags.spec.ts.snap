// Vitest Snapshot v1

exports[`n-dynamic-tags > should work with \`size\` prop 1`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(22px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgb(224, 224, 230); --n-close-icon-size: 12px; --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-border-radius: 2px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-close-icon-color-disabled: rgba(102, 102, 102, 1); --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 16px; --n-color: rgb(250, 250, 252); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 12px; --n-height: 22px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: rgb(51, 54, 57); --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`size\` prop 2`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(28px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgb(224, 224, 230); --n-close-icon-size: 14px; --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-border-radius: 2px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-close-icon-color-disabled: rgba(102, 102, 102, 1); --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 18px; --n-color: rgb(250, 250, 252); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 14px; --n-height: 28px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: rgb(51, 54, 57); --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`size\` prop 3`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(34px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgb(224, 224, 230); --n-close-icon-size: 14px; --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-border-radius: 2px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-close-icon-color-disabled: rgba(102, 102, 102, 1); --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 18px; --n-color: rgb(250, 250, 252); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 14px; --n-height: 34px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: rgb(51, 54, 57); --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`type\` prop 1`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(22px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgb(224, 224, 230); --n-close-icon-size: 12px; --n-close-color-pressed: rgba(0, 0, 0, .13); --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-border-radius: 2px; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-close-icon-color-disabled: rgba(102, 102, 102, 1); --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 16px; --n-color: rgb(250, 250, 252); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 12px; --n-height: 22px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: rgb(51, 54, 57); --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`type\` prop 2`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(22px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgba(32, 128, 240, 0.3); --n-close-icon-size: 12px; --n-close-color-pressed: rgba(32, 128, 240, 0.18); --n-close-color-hover: rgba(32, 128, 240, 0.12); --n-close-border-radius: 2px; --n-close-icon-color: #2080f0; --n-close-icon-color-hover: #2080f0; --n-close-icon-color-pressed: #2080f0; --n-close-icon-color-disabled: #2080f0; --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 16px; --n-color: rgba(32, 128, 240, 0.1); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 12px; --n-height: 22px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: #2080f0; --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`type\` prop 3`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(22px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgba(24, 160, 88, 0.3); --n-close-icon-size: 12px; --n-close-color-pressed: rgba(24, 160, 88, 0.18); --n-close-color-hover: rgba(24, 160, 88, 0.12); --n-close-border-radius: 2px; --n-close-icon-color: #18a058; --n-close-icon-color-hover: #18a058; --n-close-icon-color-pressed: #18a058; --n-close-icon-color-disabled: #18a058; --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 16px; --n-color: rgba(24, 160, 88, 0.1); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 12px; --n-height: 22px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: #18a058; --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`type\` prop 4`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(22px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgba(240, 160, 32, 0.35); --n-close-icon-size: 12px; --n-close-color-pressed: rgba(240, 160, 32, 0.18); --n-close-color-hover: rgba(240, 160, 32, 0.12); --n-close-border-radius: 2px; --n-close-icon-color: #f0a020; --n-close-icon-color-hover: #f0a020; --n-close-icon-color-pressed: #f0a020; --n-close-icon-color-disabled: #f0a020; --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 16px; --n-color: rgba(240, 160, 32, 0.12); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 12px; --n-height: 22px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: #f0a020; --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;

exports[`n-dynamic-tags > should work with \`type\` prop 5`] = `"--n-font-weight-strong: 500; --n-avatar-size-override: calc(22px - 8px); --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 2px; --n-border: 1px solid rgba(208, 48, 80, 0.23); --n-close-icon-size: 12px; --n-close-color-pressed: rgba(208, 48, 80, 0.18); --n-close-color-hover: rgba(208, 48, 80, 0.12); --n-close-border-radius: 2px; --n-close-icon-color: #d03050; --n-close-icon-color-hover: #d03050; --n-close-icon-color-pressed: #d03050; --n-close-icon-color-disabled: #d03050; --n-close-margin: 0 0 0 4px; --n-close-margin-rtl: 0 4px 0 0; --n-close-size: 16px; --n-color: rgba(208, 48, 80, 0.08); --n-color-checkable: #0000; --n-color-checked: #18a058; --n-color-checked-hover: #36ad6a; --n-color-checked-pressed: #0c7a43; --n-color-hover-checkable: rgba(46, 51, 56, .09); --n-color-pressed-checkable: rgba(46, 51, 56, .13); --n-font-size: 12px; --n-height: 22px; --n-opacity-disabled: 0.5; --n-padding: 0 7px; --n-text-color: #d03050; --n-text-color-checkable: rgb(51, 54, 57); --n-text-color-checked: #FFF; --n-text-color-hover-checkable: rgb(51, 54, 57); --n-text-color-pressed-checkable: rgb(51, 54, 57);"`;
